src/include/sof/audio/igo_nr/igo_nr_comp.h

```diff
@@ -0,0 +1,48 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2021 Intelligo Corporation. All rights reserved.
+ *
+ * Author: Fu-Yun TSUO <fy.tsuo@intelli-go.com>
+ */
+
+#ifndef __SOF_AUDIO_IGO_NR_CONFIG_H__
+#define __SOF_AUDIO_IGO_NR_CONFIG_H__
+
+#include <sof/platform.h>
+#include <sof/audio/audio_stream.h>
+#include <sof/audio/igo_nr/igo_lib.h>
+
+/* IGO_NR only support XCC HIFI3 */
+#if defined __XCC__ && XCHAL_HAVE_HIFI3 == 1
+#include <xtensa/config/core-isa.h>
+#else
+#error "No XCC/HIFI3 found. Cannot build IGO_NR module."
```

Comment:
I think we have a kconfig HiFi, please just use that for the dependency on HiFi

Reply:
It must be Hifi3 of TGL, which is only available during compiling. I've checked in Kconfig and there is no such flag.

Reply:
The wrapper should not have a hard dependency on platform, you should be able to reuse the interface and recompile your library for other platforms.

Reply:
In general the wrapper should be able to build on any platform as it's generic C. This means you only need to port your closed source binary for other architecture and the wrapper will just work.

Reply:
Agree with you about the definition of wrapper. I will change accordingly.

However if the toolchain to compile static library doesn't match the one to compile SOF framework, it might just show a plain linking error. Any suggestion to deal with this situation.

Reply:
Another point to note is that igo_nr static library might have various HiFi version but never gcc, because the MCPS will be too large without Xtensa hardware acceleration.

Reply:
I would put some text in the Kconfig help stating what ISA the binary supports today and with some link to your company. This way users can contact your company and request your product with a different ISA config etc. e.g. "This feature links to a proprietary binary X that currently is supported on xtensa ISA config Y. Please see URL for any questions about the binary".

Reply:
sounds good. thanks for the suggestion and I will update the Kconfig description.